### PR TITLE
Feat/ledgerizer setup

### PR DESCRIPTION
**Pull request by dacasas**
* Se implementa la gema Ledgerizer
* Se crean los commands para efectuar movimientos al hacer un click promocionado o al hacer un deposito.

### CODE REVIEW

File: app/commands/deposit_accounter.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class DepositAccounter < PowerTypes::Command.new(:deposit, :store, :organization, :amount)
+  include Ledgerizer::Execution::Dsl
+
+  def perform
+    execute_store_deposit_entry(tenant: @organization, document: @deposit, date: Date.today) do
```
Comment (monkeyci) [inline app/commands/deposit_accounter.rb]: Do not use Date.today without zone. Use Time.zone.today instead.

File: config/initializers/money.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  # To handle the inclusion of validations for monetized fields
+  # The default value is true
+  #
+  # config.include_validations = true
+
+  # Default ActiveRecord migration configuration values for columns:
+  #
+  # config.amount_column = { prefix: '',           # column name prefix
+  #                          postfix: '_cents',    # column name  postfix
+  #                          column_name: nil,     # full column name (overrides prefix, postfix and accessor name)
```
Comment (monkeyci) [inline config/initializers/money.rb]: Line is too long. [115/100]

File: config/initializers/money.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# encoding : utf-8
+
+MoneyRails.configure do |config|
+
```
Comment (monkeyci) [inline config/initializers/money.rb]: Extra empty line detected at block body beginning.

File: config/initializers/money.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
+# encoding : utf-8
```
Comment (monkeyci) [inline config/initializers/money.rb]: Unnecessary utf-8 encoding comment.

File: app/models/organization.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+# == Schema Information
+#
+# Table name: organizations
+#
+#  id         :bigint(8)        not null, primary key
+#  name       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
```
Comment (monkeyci) [inline app/models/organization.rb]: 1 trailing blank lines detected.

File: app/models/deposit.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+#
+#  index_deposits_on_organization_id  (organization_id)
+#  index_deposits_on_store_id         (store_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (organization_id => organizations.id)
+#  fk_rails_...  (store_id => stores.id)
+#
+
```
Comment (monkeyci) [inline app/models/deposit.rb]: 1 trailing blank lines detected.

File: app/commands/promoted_click_accounter.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class PromotedClickAccounter < PowerTypes::Command.new(:product_action, :store, :organization)
+  include Ledgerizer::Execution::Dsl
+
+  def perform
+    amount = Money.from_amount(ENV['CPC'])
+    execute_promoted_click_entry(tenant: @organization, document: @product_action, date: Date.today) do
```
Comment (monkeyci) [inline app/commands/promoted_click_accounter.rb]: Do not use Date.today without zone. Use Time.zone.today instead.<br>Line is too long. [103/100]

File: config/initializers/ledgerizer.rb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+#     end
+#
+#     conf.entry :user_deposit_distribution, document: :user_deposit do
+#       conf.debit account: :funds_to_invest, accountable: :user
+#       conf.credit account: :to_invest_in_fund, accountable: :user
+#     end
+#   end
+# end
+
+Ledgerizer.setup do |conf|
```
Comment (ldlsegovia) [inline config/initializers/ledgerizer.rb]: Mirá, podés testear la configuración así: https://github.com/platanus/fintual-rails/blob/master/spec/config/initializers/ledgerizer_spec.rb

File: app/commands/promoted_click_accounter.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+class PromotedClickAccounter < PowerTypes::Command.new(:product_action, :store, :organization)
+  include Ledgerizer::Execution::Dsl
+
+  def perform
+    amount = Money.from_amount(ENV['CPC'])
+    execute_promoted_click_entry(
+      enant: @organization,
```
Comment (ldlsegovia) [inline app/commands/promoted_click_accounter.rb]: 👀 `enant`

File: app/commands/promoted_click_accounter.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
+class PromotedClickAccounter < PowerTypes::Command.new(:product_action, :store, :organization)
+  include Ledgerizer::Execution::Dsl
+
+  def perform
+    amount = Money.from_amount(ENV['CPC'])
```
Comment (ldlsegovia) [inline app/commands/promoted_click_accounter.rb]: mejor `ENV.fetch("CPC")` para que tire exception si te olvidas de setearlo. Si se usa en varios lugares, quizás podrías tener un `cpc_price` en algún modelo o como monkey patch en en `Money`. `Money.cpc_price`

Comment (dacasas) [reply app/commands/promoted_click_accounter.rb]: Si, esa variable sera temporal de todas formas. Jaime quiere tenerlo así para poder cambiar mas rápido  el valor de los click (CPC = Cost per click) al principio, e ir probando diferentes valores me imagino.

File: spec/config/initializers/ledgerizer_spec.rb [hunk 11]
```diff
@@ -0,0 +1,128 @@
+require "rails_helper"
+
+# rubocop:disable RSpec/NestedGroups
```
Comment (monkeyci) [inline spec/config/initializers/ledgerizer_spec.rb]: Re-enable RSpec/NestedGroups cop with # rubocop:enable after disabling it.

File: spec/commands/promoted_click_accounter_spec.rb [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  it "adds lines to ledgerizer" do
+    perform
+
+    last_entry = Ledgerizer::Entry.last
+    expect(last_entry).to have_ledger_line(account_name: :clicks_service, accountable: organization,
+      amount: Money.from_amount(amount))
+    expect(last_entry).to have_ledger_line(account_name: :available_store_funds, accountable: store,
+      amount: Money.from_amount(amount * -1))
```
Comment (monkeyci) [inline spec/commands/promoted_click_accounter_spec.rb]: Align the elements of a hash literal if they span more than one line.

File: spec/commands/promoted_click_accounter_spec.rb [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+      document: product_action
+    )
+  end
+
+  it "adds lines to ledgerizer" do
+    perform
+
+    last_entry = Ledgerizer::Entry.last
+    expect(last_entry).to have_ledger_line(account_name: :clicks_service, accountable: organization,
+      amount: Money.from_amount(amount))
```
Comment (monkeyci) [inline spec/commands/promoted_click_accounter_spec.rb]: Align the elements of a hash literal if they span more than one line.

File: spec/commands/deposit_accounter_spec.rb [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  it "adds lines to ledgerizer" do
+    perform
+
+    last_entry = Ledgerizer::Entry.last
+    expect(last_entry).to have_ledger_line(account_name: :bank_account, accountable: organization,
+      amount: Money.from_amount(amount))
+    expect(last_entry).to have_ledger_line(account_name: :available_store_funds, accountable: store,
+      amount: Money.from_amount(amount))
```
Comment (monkeyci) [inline spec/commands/deposit_accounter_spec.rb]: Align the elements of a hash literal if they span more than one line.

File: spec/commands/deposit_accounter_spec.rb [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+      document: store_deposit
+    )
+  end
+
+  it "adds lines to ledgerizer" do
+    perform
+
+    last_entry = Ledgerizer::Entry.last
+    expect(last_entry).to have_ledger_line(account_name: :bank_account, accountable: organization,
+      amount: Money.from_amount(amount))
```
Comment (monkeyci) [inline spec/commands/deposit_accounter_spec.rb]: Align the elements of a hash literal if they span more than one line.

File: spec/commands/promoted_click_accounter_spec.rb [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+
+  it "adds lines to ledgerizer" do
+    perform
+
+    last_entry = Ledgerizer::Entry.last
+    expect(last_entry).to have_ledger_line(
+      account_name: :clicks_service, accountable: organization, amount: Money.from_amount(amount)
+    )
+    expect(last_entry).to have_ledger_line(
+      account_name: :available_store_funds, accountable: store, amount: Money.from_amount(amount * -1)
```
Comment (monkeyci) [inline spec/commands/promoted_click_accounter_spec.rb]: Line is too long. [102/100]

Comment (monkeyci) [inline spec/commands/promoted_click_accounter_spec.rb]: Line is too long. [102/100]

**Comment by dacasas** [comment]
[SetUp Ledgerizer](https://trello.com/c/u7xWKLzS/136-setup-ledgerizer)

**Pull request merged**